### DISCUSSION TITLE
Home Page

### DISCUSSION THREAD

**Discussion by thescientist13** [ProjectEvergreen/www.greenwoodjs.dev#4] [Ideas]
## Overview
As part of the website redesign, we will want to re-think the home page in terms of the story that it can tell about Greenwood and its philosophies.  Taking a "show" don't tell approach, I would like to highlight usage of Greenwood, as a way to emphasize and demonstrate how much we love web standards.  Aside from the traditional hero banner / CTA section, the next section I would like to focus on is a narrative driven walk-through of Greenwood highlighting features like hybrid routing / rendering, API endpoints, SSR w/ Web Components, etc.

From there, we can have feature bullets and links to explore further.  And, if possible, end it all with a nice showcase of Greenwood projects demonstrating things like serverless adapters, htmx, and general community focused integrations.
**_Mockup_**
<details>
  <img src="https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/652688a1-df03-4941-ae1e-75c5accb0da0">
</details>

### Hero
Would obviously like to have a nice pull quote and nice image, but not sure what to put or what to show off?

Current copy is
> **Your full-stack workbench for the web**
> Greenwood embraces web standards to keep the focus on your work, not the framework. It's web dev refreshed!

For some pull quotes, some thoughts are:
- The Full-Stack Web is here
- Stop using some body else’s API
- Completely independent / vendor less
- Building the web from front to back
- The evergreen web framework 

### Zero-To-Hero Walkthrough

Would like to showcase going from static to dynamic with Greenwood across a good cross section of the feature set, kind of like the Vercel mantra _Deploy, Preview, Ship_.

So thinking we could highlight these features to "build" a feature?
1. Starting with HTML
1. SSR pages
1. API Endpoints
1. Web Components

Then... deploy! 🚀 

### Feature Bullets
- HTML first
- Bundleless development
- SSR, SPA, SSG, etc
- Content as Data
- Static Asset Bundling
- Real, Full-Stack (isomorphic) Web Components (WCC)
- File Based Routing

### Community Showcase
Some examples that could help demonstrate some "social proof" could be:
- htmx + serverless - https://github.com/thescientist13/greenwood-htmx
- Lit SSR + serverless (WIP) - https://github.com/thescientist13/greenwood-demo-adapter-vercel-lit
- Something with Tailwind - https://github.com/AnalogStudiosRI/www.blissfestri.com
- TypeScript - https://github.com/AnalogStudiosRI/www.analogstudios.net
- Hackernews (TODO)
- Movies App? (TODO)

----

**Content Status**
1. [ ] Hero
    - headline / pull quote
    - image
    - CTAs
1. [ ] Walkthrough
    - steps to success and copy
    - how to demonstrate (images)
1. [ ] Features
    - how to present
    - what to say
1. [ ] Showcase
    - project types

**Comment by thescientist13** [comment]
Got a home page preview going here
https://deploy-preview-18--super-tapioca-5987ce.netlify.app/

Would still like to refine the code walk through example but I'm digging it so far.

**Comment by Auhseh** [comment]
Hello Owen,
Here’s my contribution. Will update this file with future progress

P.s I see your vision now 😄, Geist is 🔥

https://www.figma.com/file/ePhyD5qNMrBAcHwQ08BnQC/GreenWood?type=design&node-id=113%3A740&mode=design&t=400dqKMaa6XxjPvW-1

**Comment by thescientist13** [comment]
Wanted to present you this take on the home page I have been thinking of the past week in terms of content and content section, to see what you think.  Obviously this is somewhat hacked together but should provide a good reference.  I'm really like the shape of this one in particular.  (styleguide TBD)

![Home Page mockup](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/aaef721e-2fd5-4fdc-8ecd-b3baa33fb4c5)

Also, might be good to make the _build and deploy_ providers logos a bit bigger so they really catch your eye on the way down.  Eventually the hope is to support AWS and Cloudflare which will have nice bright logos!  ✨ 

General notes and thoughts guiding me to this
- Let's make sure we can try and link out to a guide or doc for most features and examples (this would be good to help encourage people to prob the website past just the home page).  Mostly on me to make the content to link to, but let's explore adding more "CTA"s like _learn more_, _see a demo_, etc in strategic areas of the content.
- Rename _The web has evolved..._ to -> **Why Greenwood?**
- Rename _Build and deploy..._ to -> _**Run anywhere the web can run**_
- Added a _Build with "friends"_ section to show off ecosystem compat (open to better heading wording here)

**Reply by thescientist13** [reply]
This will also be good motivation for us to test out Greenwood with more libraries in the ecosystem like Stencil, Lightning Web Components, and FAST so we can expand out the logos.

Not sure if it should just be one box so either side doesn't look to lopsided?

**Reply by Auhseh** [reply]
I love this Owen, this is perfect!

I’ll: 
- make the Provider logos bigger as suggested 
- Tweak the Headings as suggested
- Integrate the new libraries section into the design
- Integrate the CTAs as suggested 

But overall, I love how much progress we’ve made.

Also, +1 on testing with more libraries

**Reply by thescientist13** [reply]
One last feedback item after meeting with the team was that we think for the footer we can probably just have the links and logos, that way we can have the same footer across the site and I think we might want to save that space for the eventual showcase section.
![Screenshot 2024-03-17 at 11 34 46 AM](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/c2c7462a-fe0f-4167-a80c-862f007cafa6)

Otherwise I think this is shaping up really nicely and I am very excited to start working on it!

**Reply by thescientist13** [reply]
Also, I will play around with some of the sizing here for maybe more of 40 / 60 split between content and code?  I am ok with taking out some content if the code section can get more room, not that we have our features section on the home page.
<img width="821" alt="Screenshot 2024-03-17 at 11 49 46 AM" src="https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/8d6cb0b9-2f29-4c81-8387-834dd1f8caf1">

That was one of reasonings behind putting the content at the top, so there could be plenty of room for the code snippets.
![Screenshot 2024-03-17 at 11 52 06 AM](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/assets/895923/0bc97b85-e3b2-4532-a82e-3a5b43972cac)

But also knowing that we need to make it look good for mobile presents an interesting set of circumstances though for this treatment more broadly, so there are going to have to be scroll bars at some point I imagine?  We can chat a bit about this on our call though.

**Comment by Auhseh** [comment]
> Also, I will play around with some of the sizing here for maybe more of 40 / 60 split between content and code? I am ok with taking out some content if the code section can get more room, not that we have our features section on the home page. <img alt="Screenshot 2024-03-17 at 11 49 46 AM" width="821" src="https://private-user-images.githubusercontent.com/895923/313486377-8d6cb0b9-2f29-4c81-8387-834dd1f8caf1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3MTkyMDIsIm5iZiI6MTcxMDcxODkwMiwicGF0aCI6Ii84OTU5MjMvMzEzNDg2Mzc3LThkNmNiMGI5LTJmMjktNGM4MS04Mzg3LTgzNGRkMWY4Y2FmMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QyMzQxNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZjdmOWY3ZWMwZmFhMjg0NWZkMzliNzQ3ZjRlYmFhYWEwZGYwY2M0Nzk1NzZhYTJjMzMyODQ4ZjQwNjcyMTRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.YhL9j4SPKG7fgR26-6ulWnIw_b-bC6RE_nxzCPGvWoE">
> 
> That was one of reasonings behind putting the content at the top, so there could be plenty of room for the code snippets. ![Screenshot 2024-03-17 at 11 52 06 AM](https://private-user-images.githubusercontent.com/895923/313486531-0bc97b85-e3b2-4532-a82e-3a5b43972cac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3MTkyMDIsIm5iZiI6MTcxMDcxODkwMiwicGF0aCI6Ii84OTU5MjMvMzEzNDg2NTMxLTBiYzk3Yjg1LWUzYjItNDUzMi1hODJlLTNhNWI0Mzk3MmNhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QyMzQxNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjcwOGY1M2E3YTk4YThhYzk0YzRmZWJmNTkwZjMwOTUzMjc4ZmFkMGY1OGRmNDRhNDViZTg4MDkyNWRhYzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.R0fa-2YG7rMT9J9WunsJk7DLvnl1vz-lsfUIsDwwRho)
> 
> But also knowing that we need to make it look good for mobile presents an interesting set of circumstances though for this treatment more broadly, so there are going to have to be scroll bars at some point I imagine? We can chat a bit about this on our call though.

I played around with this in the implementation, I think that could work.

Re: Mobile, I think we can hide the Code snippets on mobile and just show the content.

**Comment by thescientist13** [comment]
Ok, moving this over to an issue now that we've got our production draft almost ready 💯 
https://github.com/ProjectEvergreen/www.greenwoodjs.dev/issues/30
